core/arch/arm/sm/sm_a32.S

```diff
@@ -314,9 +329,37 @@ UNWIND(	.fnstart)
 	sub	sp, r0, #(SM_CTX_SIZE - SM_CTX_NSEC)
 	msr	cpsr, r1
 
+#ifdef CFG_CORE_WORKAROUND_SPECTRE_BP
+	/*
+	 * For unrecognized CPUs we fall back to the vector used for
+	 * unaffected CPUs. Cortex A-15 has special treatment compared to
+	 * the other affected Cortex CPUs.
+	 */
+	read_midr r1
+	ubfx	r2, r1, #MIDR_IMPLEMENTER_SHIFT, #MIDR_IMPLEMENTER_WIDTH
+	cmp	r2, #MIDR_IMPLEMENTER_ARM
+	bne	1f
```

Comment:
minor: are these 3 instruction useful since loading `sm_vect_table` as vector table is the default behaviour for not impacted cores?

Reply:
Without this test the primary part number isn't defined. I'm skipping the architecture field as we're always compiling this for ARMv7.